syntax = "proto3";

package bi;

service BIService {
  rpc ListDocs (ListDocsRequest) returns (ListDocsResponse);

  // Incidents (new model)
  rpc QueryIncidents (QueryIncidentsRequest) returns (QueryIncidentsResponse);
  rpc AggByType (AggByTypeRequest) returns (AggByTypeResponse);
  rpc AggBySeverity (AggBySeverityRequest) returns (AggBySeverityResponse);
}

message ListDocsRequest {
  int32 limit = 1;
}

message Doc {
  int32 id = 1;
  string mapper_version = 2;
  string created_at = 3;
}

message ListDocsResponse {
  repeated Doc docs = 1;
}

// ------------------------
// QueryIncidents
// ------------------------
message QueryIncidentsRequest {
  string type = 1;
  string severity = 2;
  string status = 3;
  string country = 4;
  int32 limit = 5;
}

message Incident {
  int32 doc_id = 1;
  string incident_id = 2;
  string source = 3;
  string incident_type = 4;
  string severity = 5;
  string status = 6;

  string city = 7;
  string country = 8;
  string continent = 9;

  double lat = 10;
  double lon = 11;
  double accuracy_m = 12;

  string reported_at = 13;
  string validated_at = 14;
  string resolved_at = 15;
  string last_update_utc = 16;

  string assigned_unit = 17;
  double resources_count = 18;
  double eta_min = 19;
  double response_time_min = 20;

  double estimated_cost_eur = 21;
  double risk_score = 22;
}

message QueryIncidentsResponse {
  repeated Incident incidents = 1;
}

// ------------------------
// AggByType
// ------------------------
message AggByTypeRequest {}

message AggByTypeRow {
  string incident_type = 1;
  int32 total_incidents = 2;
  double avg_risk_score = 3;
  double total_estimated_cost_eur = 4;
}

message AggByTypeResponse {
  repeated AggByTypeRow rows = 1;
}

// ------------------------
// AggBySeverity
// ------------------------
message AggBySeverityRequest {}

message AggBySeverityRow {
  string severity = 1;
  int32 total_incidents = 2;
  double avg_risk_score = 3;
}

message AggBySeverityResponse {
  repeated AggBySeverityRow rows = 1;
}